src/main/java/com/microfocus/application/automation/tools/octane/CIJenkinsServicesImpl.java

```diff
@@ -467,27 +465,11 @@ public OctaneResponse upsertCredentials(CredentialsInfo credentialsInfo) {
 	}
 
 	private ACLContext startImpersonation() {
-		OctaneServerSettingsModel settings = ConfigurationService.getSettings(getInstanceId());
-		if (settings == null) {
-			throw new IllegalStateException("failed to retrieve configuration settings by instance ID " + getInstanceId());
-		}
-		String user = settings.getImpersonatedUser();
-		User jenkinsUser = null;
-		if (user != null && !user.isEmpty()) {
-			jenkinsUser = User.get(user, false, Collections.emptyMap());
-			if (jenkinsUser == null) {
-				throw new PermissionException(401);
-			}
-		} else {
-			logger.info("No user set to impersonating to. Operations will be done using Anonymous user");
-		}
-
-		ACLContext impersonatedContext = ACL.as(jenkinsUser);
-		return impersonatedContext;
+		return ImpersonationUtil.startImpersonation(getInstanceId());
```

Comment:
If an exception occurs, it won't be caught?

Reply:
the only change that I done here is to move "startImpersonation" method to different class. but basically, this is very old code, which been used in a lot of different contexts. so I don't want to change it now.
so for now, I caught the exception (in the fix to PR) only when I used this method, in Webhooks class.